### PR TITLE
refactor(validation): add a helper for URI validation

### PR DESCRIPTION
**Pull request by caspiano**
A slight DRY out of uri validation logic

### CODE REVIEW

File: src/placeos-models/utilities/validation.cr [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "uri"
+
+module PlaceOS::Model::Validation
+  def self.valid_uri?(uri : String) : URI?
+    return if uri.blank?
+    parsed = URI.parse(uri)
+    parsed if !(parsed.scheme.presence.nil? || parsed.host.presence.nil?)
```
Comment (dukeraphaelng) [inline src/placeos-models/utilities/validation.cr:7]:
```suggestion
    parsed unless parsed.scheme.presence.nil? || parsed.host.presence.nil?
```

File: src/placeos-models/control_system.cr [hunk 2]
```diff
@@ -84,12 +84,8 @@ module PlaceOS::Model
 
     # Validate support URI
     validate ->(this : ControlSystem) {
-      support_url = this.support_url
-      unless support_url.empty?
-        url = URI.parse(support_url)
-        url_parsed = !!(url && url.scheme && url.host)
-        this.validation_error(:support_url, "is an invalid URI") unless url_parsed
-      end
+      return if this.support_url.blank?
```
Comment (dukeraphaelng) [inline src/placeos-models/control_system.cr:87]: There is already blank check in Validation.valid_uri?

File: src/placeos-models/module.cr [hunk 3]
```diff
@@ -246,24 +246,21 @@ module PlaceOS::Model
...
       end
 
-      url = URI.parse(uri)
-      url_parsed = !!(url.host && url.scheme)     # Ensure URL can be parsed
-      self.tls = !!(url && url.scheme == "https") # Secure indication
+      # Set secure transport flag if URI defines `https` protocol
+      self.tls = URI.parse(self.uri).scheme == "https"
 
-      self.validation_error(:uri, "is an invalid URI") unless url_parsed
+      self.validation_error(:uri, "is an invalid URI") unless Validation.valid_uri?(uri)
```
Comment (dukeraphaelng) [inline src/placeos-models/module.cr:263]:
```suggestion
      self.validation_error(:uri, "is an invalid URI") unless Validation.valid_uri?(self.uri)
```

File: src/placeos-models/control_system.cr (hunk 2 above)
Comment (caspiano) [reply src/placeos-models/control_system.cr:87]: it's valid to have a blank support URL

File: src/placeos-models/module.cr (hunk 3 above)
Comment (caspiano) [reply src/placeos-models/module.cr:263]: it's not necessary to use self here

File: src/placeos-models/module.cr [hunk 4]
```diff
@@ -273,20 +270,16 @@ module PlaceOS::Model
       self.role = driver.role
       self.port = self.port || driver.default_port || 0
```
Comment (dukeraphaelng) [inline src/placeos-models/module.cr:274]:
```suggestion
       self.port ||= driver.default_port || 0
```

File: src/placeos-models/module.cr [hunk 5]
```diff
@@ -273,20 +270,16 @@ module PlaceOS::Model
...
+      self.validation_error(:port, "is invalid") unless (1..65_535).includes?(self.port)
 
-      self.tls = false if self.udp
+      self.tls = !self.udp
 
-      url = URI.parse("http://#{ip}:#{port}/")
-      url_parsed = !!(url.scheme && url.host)
-
-      self.validation_error(:ip, "address / hostname or port are not valid") unless url_parsed
+      unless Validation.valid_uri?("http://#{ip}:#{port}/")
```
Comment (dukeraphaelng) [inline src/placeos-models/module.cr:280]:
```suggestion
      unless Validation.valid_uri?("http://#{self.ip}:#{self.port}/")
```

**Comment by dukeraphaelng** [comment]
lgtm

**Pull request merged**